Examples/ObjectiveC/UserTrackingModesExample.m

```diff
@@ -95,39 +124,29 @@ - (void)viewDidLoad {
     [self.view addSubview:self.mapView];
     
     [self setupLocationButton];
+    self.mapView.userTrackingMode = MGLUserTrackingModeNone;
 }
 
 -(void)locationButtonTapped:(UserLocationButton *)sender {
     MGLUserTrackingMode mode;
-
+    
     switch (self.mapView.userTrackingMode) {
         case MGLUserTrackingModeNone:
             mode = MGLUserTrackingModeFollow;
-            [sender updateArrowStrokeColor:self.mapView.tintColor.CGColor
-                     fillColor:UIColor.clearColor.CGColor
-                       rotation:0.66];
             break;
         case MGLUserTrackingModeFollow:
             mode = MGLUserTrackingModeFollowWithHeading;
-            [sender updateArrowStrokeColor:UIColor.clearColor.CGColor
-                     fillColor:self.mapView.tintColor.CGColor
-                       rotation:0.66];
             break;
         case MGLUserTrackingModeFollowWithHeading:
             mode = MGLUserTrackingModeFollowWithCourse;
-            [sender updateArrowStrokeColor:UIColor.clearColor.CGColor
-                     fillColor:self.mapView.tintColor.CGColor
-                       rotation:0];
             break;
         case MGLUserTrackingModeFollowWithCourse:
             mode = MGLUserTrackingModeNone;
-            [sender updateArrowStrokeColor:UIColor.whiteColor.CGColor
-                     fillColor:UIColor.clearColor.CGColor
-                       rotation:0];
             break;
     }
-    
+
     [self.mapView setUserTrackingMode:mode];
+    [sender updateArrowForTrackingMode:mode];
```

Comment:
The `updateArrowForTrackingMode:` method is only available here because this is a strange single-file beast. For the typical developer that splits their classes into individual files, this will not work — this method needs to be publicly exposed.